code/modules/antagonists/changeling/abilities/regeneration.dm

```diff
@@ -174,24 +174,67 @@
 	lock_holder = FALSE
 	ignore_holder_lock = 1
 
+	incapacitationCheck()
+		return FALSE
+
 	cast(atom/target)
 		if (..())
 			return 1
-		if (tgui_alert(holder.owner,"Are we sure?","Speed Regenerate?",list("Yes","No")) != "Yes")
-			return 1
 
-		if (!src.cooldowncheck())
-			boutput(holder.owner, "<span class='alert'>That ability is on cooldown for [round((src.last_cast - world.time) / 10)] seconds.</span>")
+		var/datum/abilityHolder/changeling/aH = holder
+		if (!istype(aH))
+			boutput(holder.owner, "<span class='alert'>That ability is incompatible with our abilities. We should report this to a coder.</span>")
 			return 1
 
-		var/mob/living/carbon/human/C = holder.owner
-		if (!istype(C))
-			boutput(holder.owner, "<span class='alert'>We have no idea what we are, but it's damn sure not compatible.</span>")
+		var/mob/living/carbon/human/H = holder.owner
+		if (tgui_alert(H,"Are we sure?","Speed regen?",list("Yes","No")) != "Yes")
+			boutput(holder.owner, "<span class='notice'>We change our mind.</span>")
 			return 1
-		boutput(holder.owner, "<span class='notice'>Your skin begins reforming around your skeleton.</span>")
 
-		while(C.health < C.max_health || !C.limbs.l_arm || !C.limbs.r_arm || !C.limbs.l_leg || !C.limbs.r_leg)
-			if(isdead(C))
-				break
-			sleep(3 SECONDS)
-			changeling_super_heal_step(C)
+		H.changeStatus("changeling_speedregen", 30 SECONDS)
+		return FALSE
+
+// changeling speedregen status effect//
+/datum/statusEffect/c_regeneration
+
+	id = "changeling_speedregen"
+	name = "Speed regeneration"
+	desc = "You quickly heal the damage dealt to you."
+	icon_state = "heart+"
+	unique = TRUE
+	maxDuration = 30 SECONDS
+
+	onAdd(optional=null)
+		. = ..()
+		var/mob/living/carbon/human/H
+		if (ishuman(owner))
+			H = owner
+			boutput(H, "<span class='notice'>We start regenerating.</span>")
+			H.visible_message("<span class='alert'><B>[H]'s flesh starts moving and sliding around oddly, repairing their wounds!</B></span>")
+			return
+		else
+			owner.delStatus("changeling_speedregen")
+
+	onUpdate(timePassed)
+		var/mob/living/carbon/human/H
+		if (!ishuman(owner)) return
+		H = owner
+		if (!H.getStatusDuration("burning"))
+			H.HealDamage("All", 2, 2, 1)
+			if (prob(20))
+				H.visible_message("<span class='alert'><B>[H]'s flesh is moving and sliding around oddly!</B></span>")
+				boutput(H, "<span class='notice'>We feel our flesh knitting back together.</span>")
```

Comment:
These messages are still really spammy, better to put a cooldown on them too and maybe add some miscellaneous fleshy noises as well? The regen can be independent of the cooldown, just don't want overwhelming chat spam.